docs/installation/installing-tentacles/polling-tentacles-web-sockets.md

```diff
@@ -0,0 +1,91 @@
+description: Octopus Polling Tentacles open a connection to the Octopus server over WebSockets to ask what to do.
+position: 1
+version: "3.12"
+---
+
+Read about [(TCP) Polling Tentacles](polling-tentacles.md) before continuing.
+
+Polling Tentacles can be setup to operate over HTTPS (Secure WebSockets) instead of raw TCP sockets. The advantage is that the port can be shared with another website (e.g. IIS or Octopus itself). The downside is the setup is a little more complicated and network communications are slightly slower.
+
+If there is an available port, we recommend using [TCP Polling Tentacles](polling-tentacles.md). If only port 443 and 80 are available, it is possible to run Octopus Web UI just on 443 (HTTPS) and a TCP Polling Tentacle on port 80. Even though it is using port 80, which is by convention HTTP, the tentacle communications will still be secure and use SSL.
+
+!toc
+
+## Server Setup
+Octopus can share HTTPS ports with any application that uses the standard Windows networking library (HTTPSYS). This includes IIS, .NET apps and Octopus itself. It does not include any applications that utilise non-HTTPSYS TCP/IP or HTTP stacks. Check your product's documentation for details. SSL offloading and sharing HTTP ports is not supported.
+
+### Listen Address
+
+The first step is to select a URL listen prefix. HTTPSYS handles the initial SSL handshake and then routes the request based on the HTTP headers. This means that the request can be routed based on IP, hostname and path. See the [UrlPrefix documentation](https://msdn.microsoft.com/en-us/library/windows/desktop/aa364698(v=vs.85).aspx) for available 
+syntax and how routes are matches. 
+
+In most cases, we recommend using `+` as the host name and a unique string for path. This will ensure that address
+takes the highest precedence. For example, to listen on port 443: `https://+:443/OctopusComms`. The path should not be 
+used by the other applications listening on the port.
+
+Once selected the Octopus Server can be configured to listen on that prefix using the following commands:
+```
+.\Octopus.Server.exe service --instance OctopusServer --stop
+.\Octopus.Server.exe configure --instance OctopusServer --commsListenWebSocket https://+:443/OctopusComms
+.\Octopus.Server.exe service --instance OctopusServer --start
+```
+
+### Testing
+To confirm that the server is successfully configured, open the listen address in your browser. If you are using `+` for the host, replace that with `localhost`. For example `https://localhost:443/OctopusComms`.
+
+If you encounter a certificate warning, ignore it and continue. You should get a page titled `Octopus Server configured successfully`. 
+
+If you do not yet know the thumbprint of your SSL certificate, you can find it by using your browser's developer tools.
+
+If you get your other application or an error, check the [server log file](/reference/log-files.md) to ensure the server is listening on the
```

Comment:
instead of "If you get your other application or an error", maybe "If you get the Octopus Portal (or the other application sharing the same port) or an error".